brickflow/codegen/databricks_bundle.py

```diff
@@ -797,15 +796,8 @@ def _build_for_each_task(
         # as this is done runtime and the used does not need to specify it in the for_each decorator (maybe it should?)
 
         workflow: Optional[Workflow] = kwargs.get("workflow")
-        builder_func_mapping = {
-            NotebookTask.__name__: self._build_native_notebook_task,
-            SparkJarTask.__name__: self._build_native_spark_jar_task,
-            SparkPythonTask.__name__: self._build_native_spark_python_task,
-            RunJobTask.__name__: self._build_native_run_job_task,
-            SqlTask.__name__: self._build_native_sql_file_task,
-        }
 
-        builder_func = builder_func_mapping.get(nested_task.__class__.__name__)
+        builder_func = self._get_task_builder(task=nested_task.task_type)
```

Comment:
Not sure if this works with the current setup, the task type will be of type for_each because that's the decorator we always use. I will use this function but maybe I will need to resolve the task type given the task class